test_conformance/allocations/main.cpp

```diff
@@ -183,12 +185,16 @@ int doTest(cl_device_id device, cl_context context, cl_command_queue queue,
                  g_reduction_percentage);
         g_max_size = (size_t)((double)g_max_size
                               * (double)g_reduction_percentage / 100.0);
-        number_of_work_items = 8192 * 2;
     }
 
     // Round to nearest MB.
     g_max_size &= (size_t)(0xFFFFFFFFFF00000ULL);
 
+    // Scales the number of work-items to keep the amount of bytes processed
+    // per work-item the same.
+    number_of_work_items =
+        std::max(g_max_size / BYTES_PER_WORK_ITEM, 8192ULL * 2ULL);
```

Comment:
You could assert that `g_max_size` is a multiple of `BYTES_PER_WORK_ITEM` maybe to guard against changes to these assumptions in the future.